Chapters

williamxu opened the pull request:
Switched out Greendao for a NoSQL db, Paper https://github.com/pilgr/Paper
Implemented persistent Manga and Chapter models
Chapter list keeps track of read manga chapters
Each chapter keeps track of the most recent page in the chapter
Lots of code cleanup

williamxu committed:
switched out greendao, using Paper https://github.com/pilgr/Paper
introduced new Manga and Chapter objects

williamxu committed:
Chapter read/unread and remember most recent page per chapter
Upon reaching last page of a chapter, it will mark it as read and color the list item orange.

williamxu commented:
Two bugs:
Left to right mode might have that weird viewpager issue where the last image gets displayed instead of the first. Swiping forward and then back will pull up the right image.

When backing out of the Viewer activity, the chapter list does not automatically highlight the read chapters, if there are any. Reopening the Item Activity will update it.

akrolsmir commented on app/.gitignore:
```diff
@@ -1 +0,0 @@
-/build
```
Did you do this on purpose? Without this .gitignore a lot of temporary build files show up...

williamxu replied:
Oh wat put that back in. I was wondering why

On Sat, Jan 9, 2016, 11:46 PM Austin Chen notifications@github.com wrote:

> In app/.gitignore
> https://github.com/williamxu/MangoReader/pull/79#discussion_r49272797:
> 
> > @@ -1 +0,0 @@
> > -/build
> 
> Did you do this on purpose? Without this .gitignore a lot of temporary
> build files show up...
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/williamxu/MangoReader/pull/79/files#r49272797.

williamxu committed:
first page being displayed as last image fixed
added gitignore back in
saves most recent page correctly
set the page index for each chapter correctly

williamxu committed:
update chapter fragment with read manga

williamxu committed:
going to next chapter updates most recent page to last page, and marks chapter as read

williamxu commented on app/src/main/java/ckmah/mangoreader/UserLibraryHelper.java:
```diff
@@ -1,162 +1,113 @@
+
+        MangaEden.getMangaEdenService(activity)
+                .getMangaDetails(m.id)
+                .enqueue(new Callback<MangaEdenMangaDetailItem>() {
```
Makes a request to download all the manga metadata. This may be a performance/bandwidth issue, especially if someone repeatedly toggles the bookmark icon. Maybe should check a last updated timestamp, and only dl when > some elapsed time, otherwise skip the request?

williamxu committed:
String constants + format
